client/landing/stepper/declarative-flow/flows/site-migration-flow/site-migration-flow.ts

```diff
@@ -142,53 +118,38 @@ const siteMigration: FlowV2 = {
 			triggerGuidesForStep( flowName, currentStep, siteId );
 		}, [ flowName, currentStep, siteId ] );
 
-		// TODO - We may need to add `...params: string[]` back once we start adding more steps.
 		async function submit( providedDependencies: ProvidedDependencies = {} ) {
-			const siteSlug = ( providedDependencies?.siteSlug as string ) || siteSlugParam || '';
-			const siteId = getSiteIdBySlug( siteSlug ) || getSiteIdParam( urlQueryParams );
-
 			switch ( currentStep ) {
 				case STEPS.SITE_MIGRATION_IDENTIFY.slug: {
 					const { from, platform, action } = providedDependencies as {
 						from: string;
-						platform: string;
+						platform: ImporterPlatform;
 						action: SiteMigrationIdentifyAction;
 					};
+					const hasDestinationSite = hasSite( siteId, siteSlug );
 
-					if ( action === 'skip_platform_identification' || platform !== 'wordpress' ) {
-						if ( isHostedSiteMigrationFlow( variantSlug ?? '' ) ) {
-							// siteId/siteSlug wont be defined here if coming from a direct link/signup.
-							// We need to make sure there's a site to import into.
-							if ( ! siteSlugParam ) {
-								return navigate( paths.siteCreationPath( { from, skipMigration: true } ) );
-							}
+					if ( hasDestinationSite ) {
```

Comment:
if we already have a destination site (siteSlug or siteId) we redirect the user directly to importList or the import-or-migrate step, based on the platform,